MobileWallet/TariLib/TariLib.swift

```diff
@@ -279,11 +295,7 @@ class TariLib {
             throw TariLibErrors.privateKeyNotFound
         }
 
-        do {
-            try tariWallet?.cancelExpiredPendingTransactions()
-        } catch {
-            TariLogger.error("Failed to cancel expired pending transactions", error: error)
-        }
+        expirePendingTransactionsAfterSync()
```

Comment:
Don't know if it already exists but maybe it would make a more complete and flexible solution to have a repeating timer to expire pending txs at reasonable intervals while the app is running, it could still start only after a successful base node sync.

Reply:
Do you mean you would prefer less or more checks on expired transactions? 
The only valid time to do this check is right after we can confirm the wallet as the most up to date state for a tx. For ios this cleanup only can happen when the app is in the foreground unfortunately and then the base node should be syncing every 30s or so

Reply:
The case on my mind is that I open the app, the expiration check is made and a tx is x mins away from expiration, I keep the app on the foreground for more than x mins and the tx remains pending although it should have been expired.

Reply:
Couldn't tell if it checks for expiration on every base node sync, if that's the case then my point is invalidated.

Reply:
Oh yeah this check happens after each base node sync, so roughly every 30s while the app is open 👍

Reply:
🎉